src/app/mod.rs

```diff
@@ -240,9 +282,20 @@ impl<S: ProvableStore + 'static> Application for BaseCoinApp<S> {
     }
 
     fn commit(&self) -> ResponseCommit {
+        let mut modules = self.modules.write().unwrap();
+        for m in modules.iter_mut() {
+            m.commit().expect("failed to commit to state");
```

Comment:
What happens in the case of a partially committed set of modules? For example, if we had 4 modules, where 2 of them committed successfully and then the 3rd one failed (and panicked)?

Will the commits from modules 1 and 2 be rolled back if we restart the application?

Reply:
basecoin-rs doesn't provide any exception safety guarantees as of now. The reason for this is that we only provide a volatile in-memory data store currently. But this is definitely something we should address in the future. 
I do have some ideas for how to achieve this with minimal effort using rust's `panic_handler` to write a crash log containing pending writes. I'll try to document the idea in an issue.

Reply:
https://github.com/informalsystems/basecoin-rs/issues/32